manager/pom.xml

```diff
@@ -151,6 +191,25 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${surefire-plugin.version}</version>
+                <configuration>
+                    <systemPropertyVariables>
+                        <quarkus.datasource.db-kind>${quarkus.datasource.db-kind}</quarkus.datasource.db-kind>
```

Comment:
This way we can distinguish which profile generated the report? I was worried about that for the db-syncer using multiple maven runs.

Reply:
To work around this I have uploaded the surefire-reports simply as a different GH artifact and made it so that the test reporter job processes and uploads both as artifacts. Unfortunately the test reporter won't work until PR is merged, but you can see it in use on the Operator as an [example](https://github.com/gingersnap-project/operator/runs/10191629234).


Unfortunately I don't think we can use surefire/failsafe executions in the cache-manager like https://github.com/gingersnap-project/db-syncer/pull/56 as we need to add DB specific dependencies for each of the different runs.

Reply:
@jabolina I was looking at https://github.com/gingersnap-project/db-syncer/pull/56/files and I think we could combine the two approaches.

We could have individual maven profiles for the per DB vendor builds (i.e. just including the required maven artifacts) as well as a "all" profile that includes the dependencies for all DB vendors. The "all" profile can make use of multiple surefire executions like your db-syncer PR so that we can have a single maven run.

The downside of this is that there will be some duplication in the pom, but I don't think that's a big deal.

Reply:
I think the "all" would be really convenient for quickly running tests in the CI or locally if necessary, while the per vendor we build only when publishing. We do not plan to publish the "all" image, right?

Another thing, I think we'll need to move stuff to the failsafe plugin to run the integration tests. I am not 100% sure we can run them with surefire.

Reply:
> We do not plan to publish the "all" image, right?

Right, no need to do that.



> Another thing, I think we'll need to move stuff to the failsafe plugin to run the integration tests. I am not 100% sure we can run them with surefire.

Sure, np with that.

Reply:
The "all" approach won't work after all, as `quarkus.datasource.db-kind` still has to be set at build time before the surefire executions :slightly_frowning_face: